offload_vio/server_rx/plugin.cpp

```diff
@@ -0,0 +1,78 @@
+#include "common/plugin.hpp"
+#include "common/switchboard.hpp"
+#include "common/data_format.hpp"
+#include "common/phonebook.hpp"
+
+#include <ecal/ecal.h>
+#include <ecal/msg/protobuf/subscriber.h>
+
+#include "vio_input.pb.h"
+
+using namespace ILLIXR;
+
+class server_reader : public plugin {
+public:
+	server_reader(std::string name_, phonebook* pb_)
+		: plugin{name_, pb_}
+		, sb{pb->lookup_impl<switchboard>()}
+		, _m_imu_cam{sb->get_writer<imu_cam_type>("imu_cam")}
+    { 
+		eCAL::Initialize(0, NULL, "VIO Offloading Sensor Data Reader");
+		subscriber = eCAL::protobuf::CSubscriber
+		<vio_input_proto::IMUCamVec>("vio_input");
+		subscriber.AddReceiveCallback(
+		std::bind(&server_reader::ReceiveVioInput, this, std::placeholders::_2));
+	}
+
+	virtual void _p_one_iteration() {
+	}
+
+
+private:
+	void ReceiveVioInput(const vio_input_proto::IMUCamVec& vio_input) {	
+		
+		// Loop through all IMU values first then the cam frame	
+		for (int i = 0; i < vio_input.imu_cam_data_size(); i++) {
+			vio_input_proto::IMUCamData curr_data = vio_input.imu_cam_data(i);
+
+			std::optional<cv::Mat> cam0 = std::nullopt;
+			std::optional<cv::Mat> cam1 = std::nullopt;
+
+			if (curr_data.rows() != -1 && curr_data.cols() != -1) {
+				size_t img0_size = curr_data.rows() * curr_data.cols();
+				char *img0_buffer = new char[img0_size + 1];
+				memcpy(img0_buffer, curr_data.img0_data().c_str(), img0_size + 1);
+				img0_buffer[img0_size] = '\0';
+
+				size_t img1_size = curr_data.rows() * curr_data.cols();
+				char *img1_buffer = new char[img1_size + 1];
+				memcpy(img1_buffer, curr_data.img1_data().c_str(), img1_size + 1);
+				img1_buffer[img1_size] = '\0';
+
+				cv::Mat img0(curr_data.rows(), curr_data.cols(), CV_8UC1, img0_buffer);
+				cv::Mat img1(curr_data.rows(), curr_data.cols(), CV_8UC1, img1_buffer);
+
+				cam0 = std::make_optional<cv::Mat>(img0);
+				cam1 = std::make_optional<cv::Mat>(img1);
+			}
+
+			_m_imu_cam.put(_m_imu_cam.allocate<imu_cam_type>(
+				imu_cam_type {
+					std::chrono::time_point_cast<std::chrono::seconds>(std::chrono::system_clock::now()),
```

Comment:
Note that this conflicts with `RelativeClock` PR. Let's decide now if `RelativeClock` should be merged before or after this PR.

Reply:
"@huzi @Jeffrey Zhang did you submit the release to the ADA site. This has become urgent - would appreciate this tomorrow morning. Please let me know" 😥

Reply:
"@ huzi @ Jeffrey did you submit the release to the ADA site. This has become urgent - would appreciate this tomorrow morning. Please let me know" 😥 
- boss